dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,529 @@
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.dotcms.saml.DotSamlConstants.SAML_USER_ID;
+
+/**
+ * This interceptor encapsulates the logic for Saml
+ * Basically if there is any configuration set on apps and c
+ * @author jsanca
+ */
+public class SamlWebInterceptor implements WebInterceptor {
+
+    public static final String APPS_SAML_CONFIG_KEY   = "app-saml-config";
+    public static final String REFERRER_PARAMETER_KEY = "referrer";
+    public static final String ORIGINAL_REQUEST       = "original_request";
+
+    protected final Encryptor       encryptor;
+    protected final LoginServiceAPI loginService;
+    protected final UserAPI         userAPI;
+    protected final UserWebAPI      userWebAPI;
+    protected final HostWebAPI      hostWebAPI;
+    protected final AppsAPI         appsAPI;
+    protected final SamlWebUtils    samlWebUtils;
+    protected final IdentityProviderConfigurationFactory identityProviderConfigurationFactory;
+    protected final SamlConfigurationService             samlConfigurationService;
+    protected final SamlAuthenticationService            samlAuthenticationService;
+
+    public SamlWebInterceptor() {
+        this(APILocator.getLoginServiceAPI());
+    }
+
+    public SamlWebInterceptor(final LoginServiceAPI loginServiceAPI) {
+        this.loginServiceAPI = loginServiceAPI;
+    }
+
+    @Override
+    public Result intercept(final HttpServletRequest request,
+                            final HttpServletResponse response) throws IOException {
+
+        HttpSession session = request.getSession(false);
+
+        if (this.samlWebUtils.isByPass(request, session)) {
+
+            Logger.info(this, ()->"Using SAML by pass");
+            return Result.NEXT;
+        }
+
+        try {
+
+            if (null != session && this.isAnySamlConfigurated()) {
+
+                final Host host = hostWebAPI.getCurrentHostNoThrow(request);
+                final IdentityProviderConfiguration identityProviderConfiguration = // gets the SAML Configuration for this site.
+                        this.identityProviderConfigurationFactory.findIdentityProviderConfigurationById(
+                                host.getIdentifier());
+
+                // If idpConfig is null, means this site does not need SAML processing
+                if (null != identityProviderConfiguration && identityProviderConfiguration.isEnabled()) { // SAML is configurated, so continue
+
+                    // check if there is any exception filter path, to avoid to canApply all the logic.
+                    if (!this.checkAccessFilters(request.getRequestURI(), this.getAccessFilterArray(identityProviderConfiguration))
+                            && this.checkIncludePath(request.getRequestURI(), this.getIncludePathArray(identityProviderConfiguration))) {
+
+                        if (null == session || this.samlWebUtils.isNotLogged(request, session)) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)